### PR TITLE
AvaloniaList compile time parsing

### LINKED ISSUE #10109: Improve AvaloniaList/List compiler support
**Is your feature request related to a problem? Please describe.**

Right now, when developer sets a string value to the AvaloniaList property, compiler will emit "AvaloniaListConverter".
Which in runtime will use [TypeUtilities.TryConvert](https://github.com/AvaloniaUI/Avalonia/blob/aad35cbacfc6da06f806083c53eec6827d166d16/src/Avalonia.Base/Utilities/TypeUtilities.cs#L130) marked as unsafe for reflection, as it can fallback to it in some code paths.

Instead, compiler should attempt to parse the list in compile time, and if it was successful, emit pre-constructed collection.

**Describe the solution you'd like**

To have a flexible support, we should add an attribute, which will work as a hint for the compiler, how to separate items in the collection.
I.e., Control.Classes is separated by spaces, while AvaloniaListConverter expects comma separated items.
We can expect something like:
```csharp
[AvaloniaListAttribute(Separators = new [] { " " })]
public Classes Classes { get { } set { } }

[AvaloniaListAttribute(Separators = new [] { "," })]
public Avalonia.Collections.AvaloniaList<double>? Dashes { get { } set { } }
```

**Additional context**

It improves following APIs:
```csharp
// Shape
        public Avalonia.Collections.AvaloniaList<double>? StrokeDashArray { get { throw null; } set { } }
// TickBar/Slider
        public Avalonia.Collections.AvaloniaList<double> Ticks { get { throw null; } set { } }
// Border
        public Avalonia.Collections.AvaloniaList<double>? BorderDashArray { get { throw null; } set { } }
// DashStyle
        public Avalonia.Collections.AvaloniaList<double>? Dashes { get { throw null; } set { } }
// PolylineGeometry
        public Avalonia.Points Points { get { throw null; } set { } }
```

Right now, we already do the same for Control.Classes property and Grid row/column definitions as a special case. It can be generalized.

Currently Avalonia.Themes.Fluent has exactly one usage of AvaloniaListConverter which can be avoided completely.

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

Instead of relying on runtime parsing which would bring unsafe AvaloniaListConverter (because of TypeConverters with reflection), attempt to do it compile time for prepared types.

Initially I added this attribute for Classes property, which doesn't use ',' as a separator and ' ' is instead, but @grokys already has redone how Classes are parsed in compile time and it has a way more specific logic now.
This attribute can be removed now, if we define "," as a separator for all AvaloniaLists and not just default.

## Fixed issues

Fixes #10109

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0033524-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by kekekeks** [comment]
I wonder if it makes sense to parse into `T[]` instead of `AvaloniaList<T>` when property is `IList<T>`-typed

**Comment by maxkatz6** [comment]
@kekekeks done, also https://github.com/kekekeks/XamlX/pull/86

File: src/Avalonia.Controls/Shapes/Polyline.cs [hunk 1]
```diff
@@ -14,10 +14,15 @@ static Polyline()
             AffectsGeometry<Polyline>(PointsProperty);
         }
 
+        public Polyline()
+        {
+            Points = new Points();
+        }
```
Comment (maxkatz6) [inline src/Avalonia.Controls/Shapes/Polyline.cs:20]: Not related to this PR, but changes nullability of points property to be consistent with PolyLineSegment/Geometry.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0033565-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Comment by robloo** [comment]
Not a huge fan of going from a clear `Points` collection to `IList<Point>`. Seems like a step backwards from an API standpoint. We really shouldn't be passing around IList following other XAML framework conventions.

WPF:
https://learn.microsoft.com/en-us/dotnet/api/system.windows.shapes.polygon.points?view=netframework-4.8

**Pull request merged**